internal/zinc-classpath/src/main/scala/sbt/internal/inc/classpath/ClasspathUtilities.scala

```diff
@@ -63,8 +63,44 @@ object ClasspathUtilities {
   final val AppClassPath = "app.class.path"
   final val BootClassPath = "boot.class.path"
 
-  def createClasspathResources(classpath: Seq[File], instance: ScalaInstance): Map[String, String] =
-    createClasspathResources(classpath, instance.allJars)
+  private[sbt] def findLibrary(instance: ScalaInstance): File = {
+    instance match {
+      case i: inc.ScalaInstance => i.libraryJar
+      case _ =>
+        (instance.allJars find { x =>
+          x.getName.startsWith("scala-library")
+        }).getOrElse {
+          throw new InvalidScalaProvider(s"Couldn't find scala-library")
+        }
```

Comment:
I guess to future proof this could find the transitive dependencies of `scala-library`, which could be computed in `launcher`, but would need some extra metadata (that the scala distribuion would need to add) when loading from `scalaHome`.